### DISCUSSION TITLE
Add a default parameter to String.prototype.repeat

### DISCUSSION THREAD

**Discussion by NEG40** [nodejs/help#4877] [General]
- Suggested new syntax: String.prototype.repeat(count, **default**)
- Somewhere in the code: if (count < 0 && **default** >= 0) count = **default**;

Currently String.prototype.repeat(count) only allows a count >= 0, adding a **default** would allow count to be negative without causing a RangeError. The **default** argument should be optional and the function should throw a RangeError when count < 0 and **default** is undefined, this will allow the update to not potentially affect other code.

Only place I could find that may be the source for this function is [node/deps/v8/src/builtins/string-repeat.tq](https://github.com/nodejs/node/blob/master/deps/v8/src/builtins/string-repeat.tq) which I started dabbling in implementing this change [here](https://github.com/NEG40/node/blob/master/deps/v8/src/builtins/string-repeat.tq) but I dont really know how to test this.

**Comment by cjihrig** [comment]
Node.js does not control this. This comes from the JavaScript language and is implemented in V8.

**Reply by NEG40** [reply]
Woops ill go that way.
